smartsim/_core/launcher/dragon/dragonBackend.py

```diff
@@ -635,10 +759,10 @@ def _refresh_statuses(self) -> None:
                     for host in group_info.hosts:
                         logger.debug(f"Releasing host {host}")
                         try:
-                            self._allocated_hosts.pop(host)
+                            self._allocated_hosts.pop(host)  # todo: remove?
                         except KeyError:
```

Comment:
What is the status of this todo?

Reply:
Solved. The allocated hosts list has been updated to hold a `Set[str]` for the steps instead of a single `str`. This line was adjusted to only pop if the set is empty instead of on every step/host iteration.